src/mlpack/core/tree/cover_tree/cover_tree_impl.hpp

```diff
@@ -1765,7 +1765,10 @@ void CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::serialize(
   }
 
   // Lastly, serialize the children.
-  ar & BOOST_SERIALIZATION_NVP(children);
+  for (size_t i = 0; i < children.size(); ++i)
+  {
+    ar & CEREAL_POINTER(children.at(i));
+  }
```

Comment:
This one is a candidate for a nice wrapper macro too (I think you already have the wrapper in place).

Reply:
Yes, I will apply the vector wrapper to this file too.